src/assets/build_assets/acknowledgements.rs

```diff
@@ -0,0 +1,199 @@
+    for entry in entries {
+        let entry = match entry {
+            Ok(entry) => entry,
+            Err(_) => continue,
+        };
+
+        let path = entry.path();
+        let stem = match path.file_stem().and_then(|s| s.to_str()) {
+            Some(stem) => stem,
+            None => continue,
+        };
+
+        handle_file(&mut acknowledgements, path, stem)?
+    }
+
+    Ok(Some(acknowledgements))
+}
+
+fn handle_file(acknowledgements: &mut String, path: &Path, stem: &str) -> Result<()> {
+    if stem == "NOTICE" {
+        handle_notice(acknowledgements, path)?;
+    } else if stem.to_ascii_uppercase() == "LICENSE" {
+        handle_license(acknowledgements, path)?;
+    }
+
+    Ok(())
+}
+
+fn handle_notice(acknowledgements: &mut String, path: &Path) -> Result<()> {
+    // Assume NOTICE as defined by Apache License 2.0. These must be part of acknowledgements.
+    let license_text = read_to_string(path)?;
+    append_to_acknowledgements(acknowledgements, &license_text)
+}
+
+fn handle_license(acknowledgements: &mut String, path: &Path) -> Result<()> {
+    let license_text = read_to_string(path)?;
+
+    if include_license_in_acknowledgments(&license_text) {
+        append_to_acknowledgements(acknowledgements, &license_text)
+    } else if license_not_needed_in_acknowledgements(&license_text) {
+        Ok(())
+    } else {
+        Err(format!("ERROR: License is of unknown type: {:?}", path).into())
+    }
+}
+
+fn include_license_in_acknowledgments(license_text: &str) -> bool {
+    let markers = vec![
+        // MIT
+        "The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.",
+
+        // BSD
+        "Redistributions in binary form must reproduce the above copyright notice,",
+
+        // Apache 2.0
+        "Apache License Version 2.0, January 2004 http://www.apache.org/licenses/",
+        "Licensed under the Apache License, Version 2.0 (the \"License\");",
+    ];
+
+    license_contains_marker(license_text, &markers)
+}
+
+fn license_not_needed_in_acknowledgements(license_text: &str) -> bool {
+    let markers = vec![
+        // Public domain
+        "This is free and unencumbered software released into the public domain.",
+
+        // Special license of assets/syntaxes/01_Packages/LICENSE
+        "Permission to copy, use, modify, sell and distribute this software is granted. This software is provided \"as is\" without express or implied warranty, and with no claim as to its suitability for any purpose."
+    ];
+
+    license_contains_marker(license_text, &markers)
+}
+
+fn license_contains_marker(license_text: &str, markers: &[&str]) -> bool {
+    let normalized_license_text = normalize_license_text(license_text);
+    for marker in markers {
+        if normalized_license_text.contains(marker) {
+            return true;
+        }
+    }
+    false
```

Comment:
Could probably be something like
```suggestion
    markers.iter().any(|marker| normalized_license_text.contains(marker))
```

Reply:
Indeed, thanks for pointing it out! Fixed now.